Add tour section to homepage.

maxnuss opened the pull request:
This adds a tour of Zulip to the bottom of the homepage. The
carousel nav is commented out so that this can go live
independent of an upgrade to the latest Bootstrap 2.

**GIFs or Screenshots:**
<img width="1280" alt="screen shot 2018-05-09 at 11 13 50 am" src="https://user-images.githubusercontent.com/2905696/39831739-3c4ffc06-537a-11e8-975a-4916b29cd0cf.png">
<img width="1215" alt="screen shot 2018-05-09 at 11 13 58 am" src="https://user-images.githubusercontent.com/2905696/39831741-3c6b22ce-537a-11e8-9b62-30a0da797168.png">
<img width="364" alt="screen shot 2018-05-09 at 11 14 18 am" src="https://user-images.githubusercontent.com/2905696/39831742-3c8707b4-537a-11e8-90bc-fff532722d1b.png">
<img width="363" alt="screen shot 2018-05-09 at 11 14 23 am" src="https://user-images.githubusercontent.com/2905696/39831743-3ca606e6-537a-11e8-8689-077d3f8e662b.png">

timabbott commented:
@maxnuss thanks for working on this!  I'd like a few changes:
* Especially in the desktop size, everything feels really small in a way that makes me not really want to read it....  I think if we increased the height something like 25%, and made the screenshots significant more zoomed-in (and also the font size for the label text moderately bigger), I think it'd help a lot.  For mobile size, I think we could try making the screenshots a little bigger by making the side-margins similar to the space between the "Zulip" and "other team chat" screenshots, but that's probably the best we can do without encouraging the user to rotate their phone (which I don't want to go down the rabbit hole of quite yet).
* The slide-right carousel transitions don't feel right, in that the 1st slide to second slide transition 
* The left/right arrows for the carousel aren't centered vertically properly (they're centered in the wrong container), and also don't feel visible enough on desktop; might want to make them a bit heavier.
* We don't have the "where you are" thing, but I assume that's just because we need to upgrade that part of bootstrap?

And then some smaller things:
* The "Organized conversations" label should have two arrows, rather than just one, to make clear it's talking about all the topics.

timabbott commented:
The other thing that Rishi and I think could look a lot better is for the screenshiots to be left-justified, rather than centered.  Basically the idea would be that the Zulip left sidebar stays in the same horizontal position through the first ~4 screenshots (up through when we first do a full-page screenshot), and the label text is left-justified above that?  I think combining that with a replace-in-place transition style (fade or something) would make this feel more like a coherent tour than a series of unrelated screenshots.

rishig commented:
Thanks Max! A few more small comments:
* Page 2: Let's add a label "Topics" with two arrows pointing to the topics
* The "Zulip" and "Other team chat" labels don't feel visible enough
* I suspect left-justifying everything (including e.g. page 7) will look better with this sort of content

Also, have no idea if this is hard or not, but would be nice for the right scroll arrow to have a small animation when you first land on this section, to indicate that the user should click it / watch the tour. If it's hard we can wait till we have the carousel navigation at the bottom.

Also, not for this iteration, but we should figure out how to make the first page here enticing enough for someone to take the tour. E.g. possibly the 4th screenshot (or a screenshot of the whole app) should be the "cover page" for the tour.

maxnuss commented:
A few comments on some of these:

- Making the screenshots/text larger: I tried this at first, but the screenshots look pretty fuzzy and low-res when blown up (example below). The way to do this right would be to recreate the screenshots in Illustrator so they can be super high res, which I can do, but not in time for the conference. If you want to move ahead with the low-res screenshots enlarged let me know; I think it looks bad but I can see the argument both ways.
![image](https://user-images.githubusercontent.com/2905696/39876828-1147aa68-5429-11e8-9dc5-52396d37fad7.png)

- Where are you thing (carousel nav): this is commented out until we update Bootstrap.

- Right scroll arrow animation: I think this is easy but those are always famous last words. Stay tuned.

Everything else makes sense to me!

timabbott commented on templates/zerver/hello.html:
```diff
@@ -625,6 +625,135 @@ <h2 class="float left">Leading organizations use Zulip:</h2>
+                <li data-target="#myCarousel" data-slide-to="5"></li>
+                <li data-target="#myCarousel" data-slide-to="6"></li>
+              </ol>  -->
+            </div>
```
I hackishly added bootstrap 2.3.2 to the page as part of an upgrade attempt, and it ended up looking like this: 

![image](https://user-images.githubusercontent.com/2746074/39881629-59a33fae-5436-11e8-8d01-7e9d2a93395d.png)

Is that what you had, or did my hackery not work right?

I think basically we're fine with having `hello.html` along get bootstrap from a CDN for now until we can upgrade properly (I've determined that it's definitely a lot of work to upgrade the logged-in app, because of a bunch of typeahead fixes we've done).

maxnuss replied:
Not quite - the dots are supposed to be below the lighter tour window. I'll just link to the CDN directly on the hello page for now!

timabbott commented:
@maxnuss awesome, this looks a lot better!  A couple more requests:
* Can you left-justify the slide title text?  Something like 
![image](https://user-images.githubusercontent.com/2746074/39892952-145d5576-5457-11e8-862a-f5137a84417d.png), but done less hackishly.  I think the multi-line ones especially look a lot better left-justified, and for all the slides, it's nice for the heading text to be in one place.
* The last slide probably needs some work on text/etc.; I think we should just extract that to a separate follow-up project.  I think it's not super critical to the overall thing.
* The streams and topics are in the same font size in Zulip; so these illustrator mockups look off
![image](https://user-images.githubusercontent.com/2746074/39893146-be096506-5457-11e8-8a6c-dfbc2d26004f.png).  (Compare with a couple slides later).
* If you have time, replacing the thin arrows with an indicator that looks a bit more fancy would be great (I assume you've by now figured out that the SVG images combined a PNG screenshot with an overlay layer that's the arrows, so they should be possible to swap out).  Rishi was thinking lines with a dot at the end, but I'm happy to mostly defer to your judgement on what's cool these days for pointing to a thing :).

Otherwise lgtm.

maxnuss commented:
@timabbott just pushed an update that left-aligns the text, fixes the font-size issue in the screenshots, and changes the arrows to dotted lines. I'm not wild about the dotted lines, but they're better than the arrows wanted to get you something usable right away (I have to head out for the day soon). I have a couple other ideas that might look better that I can implement later.

maxnuss commented:
Just pushed an update to address this round of feedback. Two quick comments:

- I ended up centering everything in the final slide, as I thought it looked better than left-aligning. Let me know if you'd like me to left-align it instead.
- Still need to implement the 0 slide

maxnuss commented:
@timabbott I'm not sure why I'm getting these invalid indentation errors in the first test—I've double-checked the file and all the indention looks right to me. Any idea what's going on?

maxnuss commented:
Thanks @rishig ! Will knock these out tomorrow

maxnuss commented:
@rishig some comments on the above:

Page 1- stretched image: this looks crappy when the low-res screenshot is blown-up. As a workaround, I blew up the screenshot but also added a light Gaussian blur, which hides the fuzziness and I think looks like an intentional choice to boot. Lmk what you think.

Page 2: Agreed both that the current lines aren't ideal and that we should punt on a better solution till we can sync up in person

Page 3 - unread count for sales: fixed

Page 3 - green lines. Don't love those either. Let me play around with this a bit more, I agree the lines as currently done are one of the weakest visual elements of this so far.

Page 3 - blue unread messages box - agreed, removed.

Swapping 5 and 6 - done

Page 5 - Removing "replying is easy..." text - done

Page 6 - adjusting "56 messages ago" text - done. Made that one line an arrow too, to indicate it's pointing outside the field of view

Page 6 - Slack blue box for unreads - good catch, I recreated this from a direct Slack screenshot and it should be much more accurate now. (Though I also enlarged in 10% - good call on that)

Transitions - not sure what you mean, this looks like a cross-fade to me. Here's a video, does this not match with what you're seeing? https://drive.google.com/open?id=15JQ6klZJMWYLlhfrE9o9wFbWpnmaRZgO

Arrow keys - this is tricky in Bootstrap 2 and I don't want to risk the upgrade (even just on this one page) to 3 or 4 which have breaking changes. Let's punt for now if that's okay.

Jumping CTA when going from first to last slide: I am 99% sure I fixed this, but I thought I fixed it once before and it came back - let me know if you see it again!

adnrs96 commented:
Hey @maxnuss I pull the PR locally and looked at the lint errors. I found some of them as the linter was suggesting. I fixed them in this file hello.html (in the archive). There is also a patch/diff file inside.
[hello.tar.gz](https://github.com/zulip/zulip/files/2009862/hello.tar.gz)

What I did not fix in that file is the lint error relating to use of CDN's. Basically we rather then serving files using CDN's, we download the files and serve them ourselves. Since by the looks of it, I think you require bootstrap only which we already have downloaded and stored at `zulip/static/third/bootstrap/js/bootstrap.js`
To include the bootstrap.js you just need to go to webpack.assets.json and add `./static/third/bootstrap/js/bootstrap.js` to the landing-page bundle (since that is what hello.html loads)
For more detailed understanding see https://zulip.readthedocs.io/en/latest/subsystems/front-end-build-process.html

Hope this helps :)

timabbott commented:
@maxnuss OK, I'm good to merge the content.

For the linter errors, you should take @adnrs96's file for the indentation.  For the other thing, you'll want to add an exclude for the CDN rule next to `templates/zilencer/billing.html` in `tools/linter_lib/custom_check.py`, and then hopefully it will pass the linter cleanly.

maxnuss commented:
Indentation fixed - thanks @adnrs96 !

@timabbott, couldn't quite figure out the exclude rule and didn't want to screw anything up, so I'll leave that for you or for the next time we work together in person

timabbott commented:
Merged as e39e632591e40d5ca0364f8fe894c8ea2e21cf5a, thanks @maxnuss!  

You can look at the `tools/linter_lib/custom_check.py ` change to see how I did the exclude, for future reference :).

BTW, now that we're using `scss`, it's totally fine to use the SCSS block syntax for new components you're adding.
